### PR TITLE
fix(sequencer): fix `deliver_tx` to accept the action tree root

### PR DESCRIPTION
**Pull request by noot**
## Summary

my previous PR #209 actually broke sequencer+cometbft since `deliver_tx` is called on every tx, after `prepare_proposal`. since it was trying to deserialize every tx into `Signed`, it was failing on the action tree commitment (as it's just 32 bytes at the start of the block)

## Changes

- update the sequencer `App` to handle the first tx in a block properly, right now it just checks that it's 32 bytes. this is fine because the content of these 32 bytes is checked in `process_proposal`.

## Testing

ran it with cometbft and now works

## Related Issues

related to #153

### CODE REVIEW

**Comment by SuperFluffy** [comment]
Can we create a very simple test that calls these methods in order?

**Comment by noot** [comment]
@SuperFluffy do you mean a unit test that that calls the `Consensus` service functions the way cometbft would? I can add that

**Comment by SuperFluffy** [comment]
@noot yeah. Ideally we'd have a cometbft mock that drives our app, but just calling these functions in order seems quicker.

File: crates/astria-sequencer-types/src/sequencer_block_data.rs [hunk 1]
```diff
@@ -138,6 +138,7 @@ impl SequencerBlockData {
     /// - if the block has no data hash
     /// - if a transaction in the block cannot be parsed
     pub fn from_tendermint_block(b: Block) -> eyre::Result<Self> {
+        // TODO: this crate should not import astria_sequencer
```
Comment (joroshiba) [inline crates/astria-sequencer-types/src/sequencer_block_data.rs:141]: nit: issue pls 🙏

File: crates/astria-sequencer/src/app.rs [hunk 2]
```diff
@@ -99,11 +110,18 @@ impl App {
...
             .expect("components should not retain copies of shared state");
 
+        self.has_block_just_begun = true;
         self.apply(state_tx)
     }
 
     #[instrument(name = "App:deliver_tx", skip(self))]
     pub(crate) async fn deliver_tx(&mut self, tx: &[u8]) -> Result<Vec<abci::Event>> {
+        if self.has_block_just_begun {
+            ensure!(tx.len() == 32);
```
Comment (joroshiba) [inline crates/astria-sequencer/src/app.rs:120]: is there any other validation which can be done on this tx?

**Review by joroshiba — APPROVED** [review]
Agreed on the need for better testing, but don't see as blocking.

File: crates/astria-sequencer-types/src/sequencer_block_data.rs (hunk 1 above)
Comment (noot) [reply crates/astria-sequencer-types/src/sequencer_block_data.rs:141]: yes  #305 will update comment

File: crates/astria-sequencer/src/app.rs (hunk 2 above)
Comment (noot) [reply crates/astria-sequencer/src/app.rs:120]: unfortunately not really, however the validators already verify this value in `ProcessProposal` by reconstructing the action tree so this is fine

File: crates/astria-sequencer/src/service/consensus.rs [hunk 3]
```diff
@@ -408,4 +408,109 @@ mod test {
...
+                votes: vec![],
+            },
+            byzantine_validators: vec![],
+        };
+        consensus_service
+            .handle_request(ConsensusRequest::BeginBlock(begin_block))
+            .await
+            .unwrap();
+
+        for tx in txs {
```
Comment (SuperFluffy) [inline crates/astria-sequencer/src/service/consensus.rs:494]: this loop would have failed previously, right?

**Review by SuperFluffy — APPROVED** [review]
Looks good, thank you for the test!

File: crates/astria-sequencer/src/service/consensus.rs (hunk 3 above)
Comment (noot) [reply crates/astria-sequencer/src/service/consensus.rs:494]: yes exactly

**Pull request merged**